src/Filesystem/File.php

```diff
@@ -450,29 +450,29 @@ public function exists(): bool
     /**
      * Returns the "chmod" (permissions) of the file.
      *
-     * @return string|null Permissions for the file, or null in case of an error
+     * @return string|bool Permissions for the file, or false in case of an error
```

Comment:
bool must be false here
Same for the other cases, as true can never be returned (we only abuse false here for null).

Reply:
PHPStan doesn't seem to have blown up so far. Why would `bool` cause issues?

Reply:
Ok. We use `bool` quite extensively as it is a type while `false` is not. If tooling works better with `string|false` that sounds like a separate change.